source/common/http/conn_manager_utility.cc

```diff
@@ -101,11 +101,11 @@ void ConnectionManagerUtility::mutateRequestHeaders(
     request_headers.insertEnvoyDownstreamServiceCluster().value(config.userAgent().value());
     HeaderEntry& user_agent_header = request_headers.insertUserAgent();
     if (user_agent_header.value().empty()) {
-      user_agent_header.value(config.userAgent().value());
+      user_agent_header.value().setStatic(config.userAgent().value());
     }
 
     if (!local_info.nodeName().empty()) {
-      request_headers.insertEnvoyDownstreamServiceNode().value(local_info.nodeName());
+      request_headers.insertEnvoyDownstreamServiceNode().value().setStatic(local_info.nodeName());
```

Comment:
I'd observe that it's hard to reason about the static allocation precondition when it's indirect, i.e. you have to transitively chain through call hierarchies and validate non-locally that it is safe. You'd think it'd be possible to validate in ASSERTs that this is safe, albeit non-portably, since static data will be allocated in the data segment somewhere.